src/backend/distributed/worker/worker_data_fetch_protocol.c

```diff
@@ -482,6 +482,7 @@ worker_apply_sequence_command(PG_FUNCTION_ARGS)
 	/* run the CREATE SEQUENCE command */
 	ProcessUtility(commandNode, commandString, PROCESS_UTILITY_TOPLEVEL,
 				   NULL, None_Receiver, NULL);
+	CommandCounterIncrement();
```

Comment:
At any rate, adding the CCI call fixes it (I think the latter hypothesis is the cause), and @anarazel said we should probably have had it in the first place.